sklearn/metrics/tests/test_ranking.py

```diff
@@ -1052,3 +1080,23 @@ def test_ranking_loss_ties_handling():
     assert_almost_equal(label_ranking_loss([[1, 0, 0]], [[0.25, 0.5, 0.5]]), 1)
     assert_almost_equal(label_ranking_loss([[1, 0, 1]], [[0.25, 0.5, 0.5]]), 1)
     assert_almost_equal(label_ranking_loss([[1, 1, 0]], [[0.25, 0.5, 0.5]]), 1)
+
+
+def test_partial_roc_auc_score():
+    # Check `roc_auc_score` for max_fpr != `None`
+    y_true = np.array([0, 0, 1, 1])
+    assert_equal(roc_auc_score(y_true, y_true, max_fpr=1), 1)
+    assert_equal(roc_auc_score(y_true, y_true, max_fpr=0.001), 1)
+    assert_equal(np.isnan(roc_auc_score(y_true, y_true, max_fpr=0)), True)
+
+    y_scores = np.array([0.1,  0,  0.1, 0.01])
+    assert_equal(
+        roc_auc_score(y_true, y_scores, max_fpr=1),
+        roc_auc_score(y_true, y_scores))
+    assert_equal(roc_auc_score(y_true, y_scores, max_fpr=0.3), 0.5)
+
+    y_true, y_pred, _ = make_prediction(binary=True)
+    for max_fpr in np.linspace(0, 1, 5):
+        assert_almost_equal(
+            roc_auc_score(y_true, y_pred, max_fpr=max_fpr),
+            _partial_roc_auc_score(y_true, y_pred, max_fpr))
```

Comment:
Also test the behavior when max_fpr is not in [0, 1]. If we raise a ValueError, then test that the ValueError is raised with an informative message.